exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java

```diff
@@ -64,13 +65,17 @@
       org.slf4j.LoggerFactory.getLogger(DrillResultSetImpl.class);
 
   private final DrillConnectionImpl connection;
+  private DrillStatementImpl drillStatement = null;
   private volatile boolean hasPendingCancelationNotification = false;
 
   DrillResultSetImpl(AvaticaStatement statement, Meta.Signature signature,
                      ResultSetMetaData resultSetMetaData, TimeZone timeZone,
                      Meta.Frame firstFrame) {
     super(statement, signature, resultSetMetaData, timeZone, firstFrame);
     connection = (DrillConnectionImpl) statement.getConnection();
+    if (statement instanceof DrillStatementImpl) {
```

Comment:
The original DrillStatement threw an exception for setting the query time out, but I don't see the same for the DrillPreparedStatement, which tries to call Avatica's implementation. My testing indicates that this is effectively a No-Op. I can extend it to that as well, but I was hoping to hear back from you first.